contracts/game.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Game is Ownable {
+  IERC20 private _token;
+  address [] judges;
+
+  struct playerDetails {
+    bool isSeekingVerification;
+    bool hasBeenVerifiedToWin;
+    bool hasWon;
+    address judgeApprovedBy;
+  }
+  mapping(address => playerDetails) private players;
+
+  event PlayerSeeksVerification(address indexed _player);
+  event PlayerVerifiedToWin(address indexed _player, address indexed _judge);
+  event PlayerWon(address indexed player);
+
+  constructor(IERC20 token) {
+    _token = token;
+  }
+
+  modifier judgeOnly {
+    require(isJudge(msg.sender), "Not a judge");
+    _;
+  }
+
+  function isJudge(address judge) view public returns (bool) {
+    bool contains = false;
+    for (uint i = 0; i < judges.length; i++) {
+      if (judge == judges[i]) {
+        contains = true;
+      }
+    }
+    return contains;
+  }
+
+  function hasTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 100 * 10**18;
+  }
+
+  function hasEnoughTokens(address player) view public returns (bool) {
+    return _token.balanceOf(player) >= 1000 * 10**18;
+  }
+
+  function hasBeenVerifiedToWin(address player) view public returns (bool) {
+    return players[player].hasBeenVerifiedToWin;
+  }
+
+  function hasNotWonBefore(address player) view public returns (bool) {
+    return !players[player].hasWon;
+  }
+
+  function seekVerification() public {
+    require(hasTokens(msg.sender), "Not enough tokens");
+    players[msg.sender].isSeekingVerification = true;
+    emit PlayerSeeksVerification(msg.sender);
+  }
+
+  function win() public {
+    require(hasTokens(msg.sender), "Not enough tokens");
+    require(hasBeenVerifiedToWin(msg.sender), "Not verified by a judge");
+    require(hasNotWonBefore(msg.sender), "Have won before");
+    if (!hasEnoughTokens(msg.sender)) {
+      (bool sent, ) = (msg.sender).call{ value: 1000 * (10*18) }("");
```

Comment:
This may not be the exact error, but this is something that needs to be fixed. The smart contract is sending out ethereum here, but we’re intending to send FWEB3 tokens. Let’s update that.

So, here’s what needs to change:

1. The contract here needs to be gifted an amount of ERC20 tokens (enough to send out). You can do this by giving it an allowance using the allowance function, so that the smart contract has tokens to play with: https://ethereum.org/en/developers/tutorials/understand-the-erc-20-token-smart-contract/
2. This line needs to be updated to do something like _token.transferFrom(address(this), msg.sender, 1000 * 10**18) so that sends out the FWEB3 token.
Might clear the error, let’s see…